test/resources/knativeresources.go

```diff
@@ -180,7 +181,11 @@ func IsKnativeObsoleteResourceGone(clients *test.Clients, namespace string, obsR
 			// This is a namespaced resource
 			_, err = clients.Dynamic.Resource(gvr).Namespace(namespace).Get(resource.GetName(), metav1.GetOptions{})
 		} else {
-			// This is a clustered resource
+			// Verify all clustered resources, except CRDs and webhooks.
+			switch strings.ToLower(resource.GetKind()) {
+			case "customresourcedefinition", "validatingwebhookconfiguration", "mutatingwebhookconfiguration":
+				continue
```

Comment:
Instead of ignoring these, shouldn't we verify they still exist? I'm confused by the inclusion of webhook configs here, too. I understand why we shouldn't delete CRD's but shouldn't the webhook configs be properly deleted?

Reply:
@jcrossley3 Here is the difficulty: we get the delta, containing a validatingwebhookconfiguration. We do not know whether validatingwebhookconfiguration still exists.
If version is the only difference for validatingwebhookconfiguration, it should exist.
If name is different, it should not exist.

Reply:
For validatingwebhookconfiguration, if we install v1beta1, then upgrade to v1, v1beta1 version can still be accessed.

Reply:
This is the reason why I suggested to consider two resources the same, when `version` is the only difference in manifestival.

Reply:
I don't understand what you mean. I'd like for someone more knowledgeable about API evolution to confirm we're doing the right thing. Essentially, if i have a v1beta1 "Thing" installed and I then create a v1 "Thing" with the same name, should we delete the v1beta1 "Thing"?

/cc @pmorie @markusthoemmes

Reply:
> This is the reason why I suggested to consider two resources the same, when `version` is the only difference in manifestival.

Sure we could do that, but is it the right thing to do? If it is, then what's the point of having an `apiVersion` at all?

Reply:
Isn't that more a problem with the dynamic client than manifestival? Does some cache need to be refreshed?

Reply:
IIRC, it should be possible to just override the v1beta1 thing with the new v1 thing **without** deleting it intermittently.

Reply:
@jcrossley3 @markusthoemmes I added a TODO comment to find out the reason, for skipping the checking of CRD's and webhooks.
I am not sure if this is a bug with dynamic client, with kube v1 and v1beta1 versioning, or this is the correct behavior.

Reply:
> IIRC, it should be possible to just override the v1beta1 thing with the new v1 thing without deleting it intermittently.

That sounds right to me. I've been trying to wrap my head around GroupVersionKind/GroupKind etc recently, and boy does it get complicated.